### ISSUE TITLE
Legacy Dotcom FSE: Atomic sites don't use FSE editor

### ISSUE THREAD

**Issue by noahtallen** [Automattic/wp-calypso#41825]
While testing a recent dotcom FSE change on Atomic sites, I went through the following flow:

1. Go through horizon.wordpress.com/start/test-fse
2. Verify that the dotcom FSE editor loads when editing a page
3. Purchase a business plan with wpcom credits
4. Install a plugin (I installed Yoast SEO) to start the transfer
5. After the transfer, the dotcom FSE editor no longer loads.

I went through all of my old dotcom FSE test sites that I had previously transferred to Atomic, and none of them had FSE active any more either. In the blog RC, it said that FSE was not active because the theme was unsupported (even though the active theme was Maywood).

I would have thought we'd hear something from support about this 🤔 

Perhaps the issue is that the format of the theme name changed or something? Or maybe the FSE support sticker was removed from the theme?

**Comment by alshakero** [comment]
I tried to debug this but I'm missing so much context. I don't really know what "dotcom FSE" means exactly. When I try to edit a page, I do see this:
![image](https://user-images.githubusercontent.com/17054134/88050227-b86d0580-cb56-11ea-8732-c9d7fce1b0ba.png)

Which makes me think that FSE is enabled and working. But I can also tell that you probably mean something else that I don't correctly understand.

To be able to continue debugging, I tried using the blog RC output `FSE is not active for this site (using an unsupported theme)` as a symptom of the bug. And I think the bug is this [line](https://github.com/Automattic/wp-calypso/blob/master/apps/full-site-editing/full-site-editing-plugin/dotcom-fse/helpers.php#L52). 

The blog RC will say `supported theme` for any negative output, but the issue can be in either of these functions `is_site_eligible_for_full_site_editing()`, `is_theme_supported()` `did_insert_template_parts()`;

**Comment by noahtallen** [comment]
> Which makes me think that FSE is enabled and working

This does mean that the "FSE plugin" is enabled and working, but that is separate from "dotcom-fse". This documentation might be helpful for understanding what "dotcom-fse" means: PCYsg-rfB-p2. TLDR is that dotcom FSE is a legacy feature contained in the FSE plugin.

The code for dotcom FSE only runs if the function you linked to returns true. In short:
1. The theme is supported (see my linked documentation for a list of supported themes)
2. The site is eligible. For simple sites, a site is only eligible if it has the full-site-editing sticker. On Atomic sites, a site is only eligible if it has the a8c-fse-is-eligible blog option. (A blog option is used on Atomic since it does not support stickers locally. The sticker is transferred to an option during automated transfer.)
3. The template parts have been created as CPTs for the theme. This happens when the theme is activated on the site during creation.

The weird thing is that all of these conditions are true before the atomic transfer (e.g. right after going through the dotcom FSE signup flow: https://horizon.wordpress.com/start/test-fse/fse-themes), but not afterwards.

This makes me think that some data is not being transferred correctly, like the sticker ->> option conversion (which was kinda flakey to begin with(

**Comment by deBhal** [comment]
I checked this out on an ephemeral site, and all three were false to start with :p

You can ssh into ephemeral sites and edit in-situ, and then `tail -f /tmp/php-errors` to dump out values for debugging just like a sandbox.

I got caught out because I tried Maywood first. The Varia theme description includes `full-site-editing` in the tags([here](https://github.com/Automattic/themes/blob/master/varia/style.scss#L12)), 
so I thought any child theme would be fine, but it turns out that only only 3 themes actually set the `full-site-editing` flag:
```
 ~/a8c/themes (master) $ ag "add_theme_support.*full-site-editing"
exford/functions.php
104:		add_theme_support( 'full-site-editing' );

maywood/functions.php
104:		add_theme_support( 'full-site-editing');

shawburn/functions.php
104:		add_theme_support( 'full-site-editing' );
```
 
But I see now that the FAQ explains why so few support it.

Getting the theme right got `is_theme_supported()` and `did_insert_template_parts()` returning `true`, but `wpcomsh` is returning `false` for the `a8c-fse-is-eligible` option and disabling FSE as a result [here]( https://github.com/Automattic/wpcomsh/blob/a5cbc6de11b7677c1723002a0448a26cb5255000/feature-plugins/full-site-editing.php#L13).

If I force that on (I stuck a `return false` in `wpcomsh_maybe_disable_fse()`) then I get the edit hover as expected:

![Add_New_Page_‹_My_WordPress_Site_—_WordPress](https://user-images.githubusercontent.com/5952255/88168971-8fd03300-cc5e-11ea-9c0c-98e790f4866e.jpg)

That really just supports your sticker -> option theory, but I hope it's at least somewhat helpful.

**Comment by noahtallen** [comment]
Yep, no worries! Thanks for the investigation. I wonder if going through the steps in the original post (i.e. normal business transferred site) will result in no option existing on the remote site. I might try to check that today

**Comment by alshakero** [comment]
> I might try to check that today

I checked and the option exists in the DB.

**Comment by alshakero** [comment]
I checked the DB of a transferred site and the flag was there in the DB.

**Comment by noahtallen** [comment]
@vindl determined the cause of this here: https://github.com/Automattic/wp-calypso/issues/45287#issue-689023877

TLDR is that there is an option created when template part data is initialized which must be present to show FSE. This option is not transferred to atomic.

**Comment by noahtallen** [comment]
Our options are: 
1. transfer that option to atomic
2. come up with another way to check for that data on atomic without needing to mess with atomic transfer code

**Comment by vindl** [comment]
I'd be in favor of option 1 for consistency. Also it's not guaranteed that alternative options would be performant enough for front end rendering, where this check is also called.

Given the very low volume of these sites, I think we should balance the time investment for fixes (I considering also just CLI interventions on case by case basis). In that sense the option transfer seems like a good way to do it. It should be a 1 line addition to whitelist the option (or 7 lines, 1 for each theme if it can be done dynamically).

**Comment by inaikem** [comment]
@noahtallen, I'm checking in to see if this issue is still relevant or can be closed as part of WPcom triage. Please let me know! 🙏

**Comment by noahtallen** [comment]
Cool! I think this probably falls under @Automattic/cylon right now. I'm not sure if the issue was fixed, but i'd guess that if it has been broken for so long (over a year), it would be more confusing to fix it at this point maybe. If folks have a non-FSE site and it suddenly gets "fixed" to be FSE, that'd be confusing, I think :) 

I have also never heard any customer reports of this issue, so it's possible no one even noticed. But either way, something to consider with the upcoming dotcom-fse migration, I think.

**Comment by creativecoder** [comment]
This appears to be because of some missing options that aren't transferred to the Atomic site. To confuse things even more, the dotcom FSE editing functionality comes back if you switch themes on the Atomic site and that theme supports it.

Let's keep this open for now--we'll either close or address it, as Cylon is actively figuring out how to handle these sites with legacy FSE functionality.

**Comment by creativecoder** [comment]
> This appears to be because of some missing options that aren't transferred to the Atomic site.

Now I see that info is already in an earlier comment!

**Comment by Addison-Stavlo** [comment]
We can close this now as dotcom FSE was deprecated months ago.
